File: internal/model/player.go
package model

import "time"

type PlayerProfile struct {
	ID                   uint64     `gorm:"primaryKey;column:id"`
	UserID               uint64     `gorm:"column:user_id"`
	Nickname             string     `gorm:"column:nickname;size:64"`
	Level                uint32     `gorm:"column:level"`
	Exp                  uint32     `gorm:"column:exp"`
	Avatar               string     `gorm:"column:avatar;size:128"`
	Power                uint64     `gorm:"column:power"`
	NameUpdatedAt        *time.Time `gorm:"column:name_updated_at"`
	NameChangeDate       *time.Time `gorm:"column:name_change_date;type:date"`
	NameDailyChangeCount uint32     `gorm:"column:name_daily_change_count"`
	CreatedAt            time.Time  `gorm:"column:created_at"`
	UpdatedAt            time.Time  `gorm:"column:updated_at"`
}

func (PlayerProfile) TableName() string {
	return "player_profiles"
}

type PlayerAsset struct {
	PlayerID         uint64     `gorm:"primaryKey;column:player_id"`
	Gold             uint64     `gorm:"column:gold"`
	Diamond          uint64     `gorm:"column:diamond"`
	Stamina          uint32     `gorm:"column:stamina"`
	StaminaUpdatedAt *time.Time `gorm:"column:stamina_updated_at"`
	CreatedAt        time.Time  `gorm:"column:created_at"`
	UpdatedAt        time.Time  `gorm:"column:updated_at"`
}

func (PlayerAsset) TableName() string {
	return "player_assets"
}

File: frontend/web/src/types/player.ts
/**
 * internal/service/player_service.go
 * 注意：/player/profile 同时输出 snake_case 与 PascalCase 两套遗留字段，
 * 前端只建模 snake_case 一套。
 */

export interface ProfileView {
  unique_id: number
  player_id: number
  user_id: number
  name: string
  nickname: string
  level: number
  exp: number
  avatar: string
  power: number
  created_at: string
  updated_at: string
}

export interface AssetView {
  player_id: number
  gold: number
  diamond: number
  stamina: number
  max_stamina: number
  next_stamina_seconds: number
  stamina_recover_every: number
}

/**
 * /heroes/star-up 返回的 assets 内嵌的是无 json tag 的 model.PlayerAsset，
 * 序列化为 PascalCase —— 全项目唯一的字段风格例外，经 normalizeAssets 归一化。
 */
export interface RawStarUpAssets {
  PlayerID: number
  Gold: number
  Diamond: number
  Stamina: number
  StaminaUpdatedAt: string | null
  CreatedAt: string
  UpdatedAt: string
}

export interface UpdateNamePayload {
  name: string
}
